umm-spec-lib/src/cmr/umm_spec/umm_to_xml_mappings/dif10.clj

```diff
@@ -223,6 +223,37 @@
   (when product-level-id
     (product-levels (str/replace product-level-id #"Level " ""))))
 
+(defn generate-dataset-citation
+  "Returns the dif10 Data_Set_Citations from UMM-C."
+  [c]
+  (let [doi (get-in c [:DOI :DOI])
+        version (:Version c)]
+    (if (empty? (:CollectionCitations c))
+      [:Dataset_Citation
+       [:Version version]
+       (when (seq doi)
+         [:Persistent_Identifier
+          [:Type "DOI"]
+          [:Identifier doi]])]
+      (for [collection-citation (:CollectionCitations c)]
+        [:Dataset_Citation
+         [:Dataset_Creator (:Creator collection-citation)]
+         [:Dataset_Editor (:Editor collection-citation)]
+         [:Dataset_Title (:Title collection-citation)]
+         [:Dataset_Series_Name (:SeriesName collection-citation)]
+         [:Dataset_Release_Date (:ReleaseDate collection-citation)]
+         [:Dataset_Release_Place (:ReleasePlace collection-citation)]
+         [:Dataset_Publisher (:Publisher collection-citation)]
+         [:Version version]
```

Comment:
DIF 10 Dataset_Citation/Version should come from the UMM ResourceCitation/Version  and not the main element version.